lib/src/configuration.cc

```diff
@@ -30,33 +30,34 @@ namespace lth_log = leatherman::logging;
 
 #ifdef _WIN32
     namespace lth_w = leatherman::windows;
-    static const fs::path DATA_DIR = []() {
+    static fs::path getWinPath(int folder_csidl) {
         wchar_t szPath[MAX_PATH+1];
-        if (FAILED(SHGetFolderPathW(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
+        if (FAILED(SHGetFolderPathW(NULL, folder_csidl, NULL, 0, szPath))) {
             throw Configuration::Error {
                 (boost::format("failure getting Windows AppData directory: %1%")
                     % lth_w::system_error()).str() };
         }
-        return fs::path(szPath) / "PuppetLabs" / "pxp-agent";
-    }();
+        return fs::path { szPath };
+    }
+
+    static const fs::path DATA_DIR = getWinPath(CSIDL_COMMON_APPDATA)
+                                        / "PuppetLabs" / "pxp-agent";
+    static const fs::path BIN_DIR = getWinPath(CSIDL_PROGRAM_FILES)
+                                        / "Puppet Labs" / "Puppet" / "pxp-agent";
```

Comment:
I recall @kylog had some objections to relative paths, but I'm not sure what they were exactly.  On *nix the pathing structure is not configurable, and there are standard / well-documented / uniform paths - so I can understand why that's the better approach there... given I don't believe there's a difference between "default" and "actual"

However, in the Windows installer we have the ability to install applications anywhere, so we probably shouldn't use `CSIDL_PROGRAM_FILES` at all since that would only work with the defaults.

Anything under `CSIDL_COMMON_APPDATA` (aka `C:\ProgramData`) is not configurable (aside from at the OS level, where the path may live on a different volume), so those paths can remain fixed.

But anything in the spec at https://github.com/puppetlabs/puppet-specifications/blob/master/file_paths.md that lives under `Program Files` should be referred to via relative path from the `.exe` given the general simplicity of that approach (i.e nothing hardcoded, nor any registry lookups required)